perllib/FixMyStreet/Queue/Item/Report.pm

```diff
@@ -216,13 +216,44 @@ sub _send {
 
     # Multiply results together, so one success counts as a success.
     my $result = -1;
+    my $report = $self->report;
+
+    my @add_send_fail_body_ids;
+    my @remove_send_fail_body_ids;
+
+    for my $sender_key ( keys %{ $self->reporters } ) {
+        my $sender = $self->reporters->{$sender_key};
+
+        # Skip if no body ID.
+        # (Lack of body at this stage shouldn't actually be possible,
+        # presumably).
+        my $body_id;
+        $body_id = $sender->bodies->[0]->id
```

Comment:
Hmm, I've just thought of an issue. As this line implies, a sender can have multiple bodies (all the bodies for the report associated with a particular SendReport class). Say a report was sent to two bodies, both with email, or both with Open311 - there'd be only one sender covering both bodies, but this would only deal with one of them. (Historically, this is for e.g. if a report was sent to two bodies both by email, it would send one email with both in the To line. It doesn't make as much sense for eg the Open311 sender which has to loop through the bodies again anyway.)

We'd really want separate results for two Open311 senders in this situation, I think (say one succeeded, one failed - we've not yet been in this situation, the common case so far is one Open311, one email), so perhaps the easiest way to deal with this is to have a different reporter for each body, not only one per sender (perhaps with an email exception for the multiple-email case mentioned above).

Reply:
I believe I have now handled this (including the email exception); not entirely confident about it though.